### ISSUE TITLE
Type error when using groupBy after mutate

### ISSUE THREAD

**Issue by fernandoabolafio** [pbeshai/tidy#59]
See the code example here: https://codesandbox.io/s/divine-cache-x3rzjj?file=/src/App.tsx

Basically doing this: 

```ts
tidy(data, mutate({ ab: (d) => d.a * d.b }), groupBy("ab"));
```

Results in a Typescript error.  

Note: It doesn't matter which argument is passed to `groupBy`, the error persists either way.

**Comment by ado387** [comment]
I found out that you can bypass this error by wrapping `mutate` and `groupBy` inside `when` clause:

```typescript
tidy(
  data,
  when( true, [ // < don't forget the bracket
    mutate( /* some mutation */ ),
    groupBy( /*some grouping */),
  ]);
);
```

While this works, I didn't have time to work out why it works, if someone can elaborate that would be great!

**Comment by pbeshai** [comment]
Should be fixed in 2.5.0+ https://codesandbox.io/s/silly-beaver-db0t60?file=/src/App.tsx
